docker-qgis/apply_deltas.py

```diff
@@ -237,8 +237,9 @@ def get_geometry_from_delta(
             if geometry.isNull():
                 raise DeltaException(f"Null geometry from {wkt=}")
 
+            # E.g. Shapefile might report a `Polygon` geometry, even though it is a `MultiPolygon`
             if geometry.type() != layer.geometryType():
-                raise DeltaException(
+                logger.info(
```

Comment:
It's not so much that QGIS wrongly reports geometry, is that's some providers can tolerate feature geometry that doesn't match its overall stated geometry. In any case, good! :)